src/libraries/System.Private.Xml/src/System/Xml/Core/XmlReader.cs

```diff
@@ -600,7 +534,7 @@ public virtual object ReadElementContentAs(Type returnType, IXmlNamespaceResolve
                 return value;
             }
 
-            return (returnType == typeof(string)) ? string.Empty : XmlUntypedStringConverter.Instance.FromString(string.Empty, returnType, namespaceResolver);
+            return returnType == typeof(string) ? string.Empty : XmlUntypedStringConverter.Instance.FromString(string.Empty, returnType, namespaceResolver!);
```

Comment:
why is `!` needed now?

Reply:
Resharper gives warning Possible 'null' assignment to entity.
Next pr I plan to replace to XmlUntypedConverter.Untyped.ChangeType(string.Empty, returnType, namespaceResolver) as done in [(https://github.com/dotnet/runtime/blob/aa2af309a12d31dd68de9dc3646d77b6f25d9d40/src/libraries/System.Private.Xml/src/System/Xml/Core/XmlReaderAsync.cs#L106)]

Reply:
If necessary, can I remove it?

Reply:
@kronic, I assume if it wasn't needed before it shouldn't be now, especially namespaceResolver is non-nullable in that context. For next PR, let's talk about it in next PR, this is likely ok but prefer to think about it separately (also easier to track down if we regress something)